From 88b9a93d61824ba8b18cc9fc943527937db5e40b Mon Sep 17 00:00:00 2001
From: Aaron Muir Hamilton <aaron@correspondwith.me>
Date: Fri, 17 Nov 2017 12:27:30 +0900
Subject: [PATCH 5/5] src: Reintroduce the hex mode keybind with an environment
 variable

  This was removed by "Move emoji implementation from
  IBusEngineSimple to IBusPanel". Reintroduced here
  behind an environment variable.

BUG=https://github.com/ibus/ibus/issues/1961

Review URL: https://codereview.appspot.com/335030043

Patch from Aaron Muir Hamilton <aaron@correspondwith.me>.
---
 src/ibusenginesimple.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ibusenginesimple.c b/src/ibusenginesimple.c
index 6100eb7b..94ce53b7 100644
--- a/src/ibusenginesimple.c
+++ b/src/ibusenginesimple.c
@@ -80,6 +80,7 @@ struct _IBusEngineSimplePrivate {
     gchar              *tentative_emoji;
     gint                tentative_match_len;
 
+    guint               hex_mode_enabled : 1;
     guint               in_hex_sequence : 1;
     guint               in_emoji_sequence : 1;
     guint               modifiers_dropped : 1;
@@ -150,6 +151,9 @@ static void
 ibus_engine_simple_init (IBusEngineSimple *simple)
 {
     simple->priv = IBUS_ENGINE_SIMPLE_GET_PRIVATE (simple);
+    simple->priv->hex_mode_enabled =
+        g_getenv("IBUS_ENABLE_CTRL_SHIFT_U") != NULL ||
+        g_getenv("IBUS_ENABLE_CONTROL_SHIFT_U") != NULL;
 }
 
 
@@ -1075,6 +1079,7 @@ ibus_engine_simple_process_key_event (IBusEngine *engine,
         have_hex_mods = (modifiers & (HEX_MOD_MASK)) == HEX_MOD_MASK;
     }
 
+    is_hex_start = (keyval == IBUS_KEY_U) && priv->hex_mode_enabled;
     is_hex_end = (keyval == IBUS_KEY_space ||
                   keyval == IBUS_KEY_KP_Space ||
                   keyval == IBUS_KEY_Return ||
-- 
2.15.1

